js/hang-ui/src/Components/publish/CameraSourceButton.tsx

```diff
@@ -0,0 +1,50 @@
+import { Show, useContext } from 'solid-js';
+import MediaSourceSourceSelector from './MediaSourceSelector';
+import { PublishUIContext } from './PublishUIContextProvider';
+
+export default function CameraSourceButton() {
+    const context = useContext(PublishUIContext);
+    const onClick = () => {
+        const hangPublishEl = context?.hangPublish();
+        if (!hangPublishEl) return;
```

Comment:
to avoid having to do this check in a bunch of places, we could move the action into the Context, like `context?.toggleCameraSource()`. Or maybe there's a better way todo this directly with Signals?